x-pack/plugins/security_solution/public/timelines/store/middlewares/timeline_note.test.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createMockStore, kibanaMock } from '../../../common/mock';
+import { selectTimelineById } from '../selectors';
+import { TimelineId } from '../../../../common/types/timeline';
+import { persistNote } from '../../containers/notes/api';
+import { refreshTimelines } from './helpers';
+
+import {
+  startTimelineSaving,
+  endTimelineSaving,
+  showCallOutUnauthorizedMsg,
+  addNote,
+  addNoteToEvent,
+} from '../actions';
+import { updateNote } from '../../../common/store/app/actions';
+import { createNote } from '../../components/notes/helpers';
+
+jest.mock('../actions', () => {
+  const actual = jest.requireActual('../actions');
+  const endTLSaving = jest.fn((...args) => actual.endTimelineSaving(...args));
+  (endTLSaving as unknown as { match: Function }).match = () => false;
+  return {
```

Comment:
Is it worth creating a shared mock of the start and end saving that can be used by all the tests here?

Reply:
I tried moving this piece of code to `test_helpers.ts` and exporting it as a function `mockTimelineSavingAndCalloutActions`:

```typescript
export function mockTimelineSavingAndCalloutActions() {
  jest.mock('../actions', () => {
    const actual = jest.requireActual('../actions');
    const endTLSaving = jest.fn((...args) => actual.endTimelineSaving(...args));
    (endTLSaving as unknown as { match: Function }).match = () => false;
    return {
      ...actual,
      showCallOutUnauthorizedMsg: jest
        .fn()
        .mockImplementation((...args) => actual.showCallOutUnauthorizedMsg(...args)),
      startTimelineSaving: jest
        .fn()
        .mockImplementation((...args) => actual.startTimelineSaving(...args)),
      endTimelineSaving: endTLSaving,
    };
  });
}
```

However, that approach doesn't work and I think it has to do with jest needing to hoist the mocks since we're using `actual` also.

Is there a different approach you might have had in mind that might work in this scenario?

Reply:
Yea, you could use a `__mocks__` folder to handle it [described here](https://jestjs.io/docs/manual-mocks#mocking-user-modules), but that may break other tests that might mock these slightly differently. So maybe worth leaving as it is?

Reply:
Yep, I'd rather not break it for others. The overhead is minimal imo.